### PR TITLE
Support new EMP features

### PR DESCRIPTION
**Pull request by nicholaspai**
Support updates `#2`, `#3`, and `#5` from this [UMIP](https://github.com/UMAprotocol/UMIPs/pull/51/) that will update EMP contracts going forwards.

These changes to the dApp must be backwards compatible with existing EMP contracts. All older EMP's are added to a `legacyEMPs` list which removes new functionality. This makes these features available by default for all future EMP's.

Images
- Expiry menu for new EMP's
Screen Shot 2020-09-18 at 16 05 12
- New withdrawal page for expired new EMP's
Screen Shot 2020-09-18 at 16 05 17
- New message for withdrawal page for expired EMP's if there aren't any withdrawals pending
Screen Shot 2020-09-18 at 16 05 57

### CODE REVIEW

File: features/emp-selector/EMPs.ts [hunk 1]
```diff
@@ -7,7 +7,8 @@ export const EMPs: { [networkId: number]: string[] } = {
     "0xc0b19570370478EDE5F2e922c5D31FAf1D5f90EA", // uUSDrBTC-OCT
   ],
   42: [
-    "0xFb70A4CBD537B36e647553C279a93E969b041DF0", //"Perpetual", yUSD Kovan Sep30
+    "0x3366b8549047C66E985EcC43026ceD3E831e46A9", // uUSDrBTC Kovan Sep20
```
Comment (nicholaspai) [inline features/emp-selector/EMPs.ts:10]: Deployed a new EMP that expires  soon after deployment so I can test out redemptions and cancel withdrawal requests post-expiry

**Review by daywiss — APPROVED** [review]

File: features/manage-position/Create.tsx [hunk 2]
```diff
@@ -98,6 +101,7 @@ const Create = () => {
   };
 
   if (
+    network !== null &&
```
Comment (chrismaree) [inline features/manage-position/Create.tsx:104]: ooc why this extra check?

File: features/manage-position/Create.tsx [hunk 3]
```diff
@@ -161,9 +165,13 @@ const Create = () => {
       parseFloat(pricedResultantCR) >= 0 &&
       parseFloat(pricedResultantCR) < collReqFromWei;
     const transactionCRBelowGCR = transactionCR < gcr;
+    const resultantCRBelowGCR = resultantCR < gcr;
+    const cannotMint = legacyEMPs[network.chainId].includes(emp.address)
+      ? transactionCRBelowGCR
+      : transactionCRBelowGCR && resultantCRBelowGCR;
```
Comment (chrismaree) [inline features/manage-position/Create.tsx:171]: this is clean. it mimics the requires which is great.

File: features/manage-position/Create.tsx [hunk 4]
```diff
@@ -43,6 +45,7 @@ const MinLink = styled.div`
 const { formatUnits: fromWei, parseBytes32String: hexToUtf8 } = utils;
 
 const Create = () => {
+  const { network } = Connection.useContainer();
   const { contract: emp } = EmpContract.useContainer();
   const { empState } = EmpState.useContainer();
   const {
```
Comment (chrismaree) [inline features/manage-position/Create.tsx:51]: I am unable to comment below but should this PR not also change the `setTokensToMax` function to accommodate the UMIP-14 changes? the `max` amount of tokens to mint for a given amount of collateral should change accordingly, right?

**Review by chrismaree — COMMENTED** [review]
this looks great! nice and simple. I have a few questions that I left comments for.

Also, are there no changes needed to be made to the redemption page?

File: features/manage-position/Create.tsx (hunk 2 above)
Comment (nicholaspai) [reply features/manage-position/Create.tsx:104]: we need to read `network` later to fetch EMPS[network.id]

File: features/manage-position/Create.tsx (hunk 4 above)
Comment (nicholaspai) [reply features/manage-position/Create.tsx:51]: oh great point. I'll fix

**Comment by nicholaspai** [comment]
> I am unable to comment below but should this PR not also change the setTokensToMax function to accommodate the UMIP-14 changes? the max amount of tokens to mint for a given amount of collateral should change accordingly, right?

Latest change to the `setMax` and `setMin` functions which really improves UX for user's who's current CR > GCR:
- Legacy EMP's min amount of collateral needed to back +5000 tokens minted
Screen Shot 2020-09-21 at 15 14 29

- New EMP's min amount of collateral needed to back + 5000 tokens minted
Screen Shot 2020-09-21 at 15 14 42

- Legacy EMP's max amount of tokens to mint from 1 BTC collateral
Screen Shot 2020-09-21 at 15 15 10


- New EMP's max amount of tokens to mint from 1 BTC collateral
Screen Shot 2020-09-21 at 15 14 58

**Comment by nicholaspai** [comment]
> this looks great! nice and simple. I have a few questions that I left comments for.
> 
> Also, are there no changes needed to be made to the redemption page?

@chrismaree no changes to redemption page I believe, only change is to the `index.tsx` page which decides whether to render Redeem component for expired EMP's

**Review by chrismaree — APPROVED** [review]
This is great! Thanks for the hard work 🔥

**Pull request merged**